src/query_param.c

```diff
@@ -55,6 +55,7 @@ QueryParam *NewNumericFilterQueryParam_WithParams(struct QueryParseCtx *q, Query
 }
 
 void QueryParam_Free(QueryParam *p) {
+  if (!p) return;
```

Comment:
Both :)
I started calling it after some checks without checking if it was `NULL` for a simple and unified flow, but I also think that in some cases it could've happened.
Maybe I should try to cause it and see if it's an actual bug

Reply:
Verified. This couldn't happen until now